Main.c

```diff
@@ -0,0 +1,72 @@
+//Attachments area
+#include<stdio.h>
+#include<string.h>
+#include<stdlib.h>
+//print Header information
+void printHeader(int buffer)
+{
+printf("%02X\n",buffer);
+}
+
+
+void main()
+{
+/*a filename to store the file*/
+FILE *myfile;
+FILE *output;
+//i to check how many times bytes are read,o to read and print the buffer
+int i=0;
+unsigned int buffer=0;
+int o, count=0;
+char filename[50];
+    //printf("Enter the Filename : ");
+    scanf("%s",&filename);
+    myfile=fopen(filename,"rb");
+do{
+    o=fread(&buffer, 1, 1, myfile);
+    printf("%d: ",o); 
+    if(o!=1)
+    break;  
+    if (i<4)
+    {
+    
+    printf("%d: ",i);  
+    i++;
+	}
+	printHeader(buffer);
+	count++;
+  }
+   while(count<4);
+   if (buffer && 0xff000000)
```

Comment:
The condition is not right, you have used Logical & instead of bit operator and.
Here we need to know if there is any value in the sync byte area.